### PR TITLE
jms topic sender

### PR DESCRIPTION
**Pull request by gogit**
A JMS topic sender that can send one or more messages in sequence or as a parallel stream implementing the AutoCloseable interface.

### CODE REVIEW

File: test-utils/test-utils-core/src/main/java/uk/gov/justice/services/test/utils/core/messaging/TopicSender.java [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+                try {
+                    pair.getLeft().send(pair.getRight().createTextMessage(m));
+                } catch (JMSException e) {
+                    throw new RuntimeException(format("Failed to send %s", m), e);
+                }
+            });
+        });
+    }
+
+    private void exec(Consumer<Pair<MessageProducer, Session>> op) throws JMSException {
```
Comment (purple52) [inline test-utils/test-utils-core/src/main/java/uk/gov/justice/services/test/utils/core/messaging/TopicSender.java]: Could this use a BiConsumer instead?

File: test-utils/test-utils-core/src/main/java/uk/gov/justice/services/test/utils/core/messaging/TopicSender.java [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+    }
+
+
+    @Override
+    public void close() {
+        if (connection != null) {
+            try {
+                connection.close();
+            } catch (JMSException e) {
+                // ignore this as its out of our control
```
Comment (purple52) [inline test-utils/test-utils-core/src/main/java/uk/gov/justice/services/test/utils/core/messaging/TopicSender.java]: Should we wrap it in a runtime exception and rethrow?

File: test-utils/test-utils-core/src/main/java/uk/gov/justice/services/test/utils/core/messaging/TopicSender.java (hunk 1 above)
Comment (gogit) [inline test-utils/test-utils-core/src/main/java/uk/gov/justice/services/test/utils/core/messaging/TopicSender.java]: can do, will make the change :)

File: test-utils/test-utils-core/src/main/java/uk/gov/justice/services/test/utils/core/messaging/TopicSender.java (hunk 2 above)
Comment (gogit) [inline test-utils/test-utils-core/src/main/java/uk/gov/justice/services/test/utils/core/messaging/TopicSender.java]: will do

**Comment by skiddykong** [comment]
Do we need to rerun CI on this?

**Comment by purple52** [comment]
Something odd happening with Coveralls on this changeset - Travis build is still failing. We need to fix this before it can be merged in.

File: test-utils/test-utils-core/src/main/java/uk/gov/justice/services/test/utils/core/messaging/TopicSender.java [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+ */
+public class TopicSender implements AutoCloseable {
+
+    private final String uri;
+
+    private final String topic;
+
+    private Connection connection;
+
+    private TopicSender(String uri, String topic) {
```
Comment (matt-rich) [inline test-utils/test-utils-core/src/main/java/uk/gov/justice/services/test/utils/core/messaging/TopicSender.java]: Need to add final for each parameter, and also on applicable variables.

**Comment by gogit** [comment]
Adding the functionality into vagrant-port-collision branch

**Pull request closed without merging**